pkg/filter/authentication/auth.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package authentication
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+
+	"github.com/eclipse/paho.mqtt.golang/packets"
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/pipeline"
+)
+
+const (
+	// Kind is the kind of Authentication
+	Kind = "Authentication"
+
+	resultAuthFail = "AuthFail"
+)
+
+var errAuthFail = errors.New(resultAuthFail)
+
+func init() {
+	pipeline.Register(&Authentication{})
+}
+
+type (
+	// Authentication is used to check authentication for MQTT client
+	Authentication struct {
```

Comment:
This is Authentication for MQTT only so would `MQTTClientAuth`, `MQTTAuthentication` etc be better name? `pkg/filter/validator` implements currently auth* methods for HTTP and at some point it could support MQTT protocol too.

Reply:
ok, i will update it!